pkg/bootstrapper/bootstrapper.go

```diff
@@ -336,50 +335,42 @@ func (wmcb *winNodeBootstrapper) parseIgnitionFileContents(ignitionFileContents
 
 	// Find the kubelet systemd service specified in the ignition file and grab the variable arguments
 	// TODO: Refactor this to handle environment variables in argument values
+	var kubeletUnit *ignitionCfgv3Types.Unit
 	for _, unit := range configuration.Systemd.Units {
-		if unit.Name != kubeletSystemdName {
-			continue
-		}
-
-		if unit.Contents == nil {
-			return fmt.Errorf("could not process %s: Unit is empty", unit.Name)
+		if unit.Name == kubeletSystemdName {
+			kubeletUnit = &unit
+			break
 		}
+	}
+	if kubeletUnit == nil {
+		return errors.Errorf("ignition missing kubelet systemd unit file")
+	}
+	args, err := wmcb.parseKubeletArgs(*kubeletUnit)
+	if err != nil {
+		return errors.Wrap(err, "error parsing kubelet systemd unit args")
+	}
 
-		results := cloudProviderRegex.FindStringSubmatch(*unit.Contents)
-		if len(results) == 2 {
-			wmcb.kubeletArgs["cloud-provider"] = results[1]
-		}
-
-		// Check for the presence of "--cloud-config" option and if it is present append the value to
-		// filesToTranslate. This option is only present for Azure and hence we cannot assume it as a file that
-		// requires translation across clouds.
-		results = cloudConfigRegex.FindStringSubmatch(*unit.Contents)
-		if len(results) == 2 {
-			cloudConfFilename := filepath.Base(results[1])
-
-			// Check if we were able to get a valid filename. Read filepath.Base() godoc for explanation.
-			if cloudConfFilename == "." || os.IsPathSeparator(cloudConfFilename[0]) {
-				return fmt.Errorf("could not get cloud config filename from %s", results[0])
-			}
-
-			filesToTranslate[results[1]] = fileTranslation{
-				dest: filepath.Join(wmcb.installDir, cloudConfFilename),
-			}
-
-			// Set the --cloud-config option value
-			wmcb.kubeletArgs[cloudConfigOption] = filepath.Join(wmcb.installDir, cloudConfFilename)
+	// Check for the presence of "--cloud-config" option and if it is present append the value to
+	// filesToTranslate. This option is only present for Azure and hence we cannot assume it as a file that
+	// requires translation across clouds.
+	if cloudConfigPath, ok := args[cloudConfigOption]; ok {
```

Comment:
Shouldn't this now be done in `parseKubeletArgs()` and have one spot where there args are being created?

Reply:
I added a TODO explaining why it is currently like that and link to the new refactor ticket https://issues.redhat.com/browse/WINC-670

Reply:
@saifshaikh48 ticket that you asked for^